cql3/statements/raw/select_statement.hh

```diff
@@ -150,7 +151,10 @@ private:
     bool is_reversed(const schema& schema) const;
 
     /** If ALLOW FILTERING was not specified, this verifies that it is not needed */
-    void check_needs_filtering(const restrictions::statement_restrictions& restrictions);
+    void check_needs_filtering(
+            const restrictions::statement_restrictions& restrictions,
+            db::tri_mode_restriction_t::mode strict_allow_filtering,
```

Comment:
not need to do anything about this, certainly not now, but it's unfortunate that we ended up with two different meanings of the word "restriction" in the same statement.